### ISSUE TITLE
[0.9.0] Write via line protocol causing panic

### ISSUE THREAD

**Issue by consense** [influxdb/influxdb#2940]
Two writes (seperate http requests)

    snapshots_Dgbw1,user=123abc duration=5207,productivity=-0.02,pSeconds=-0.10414 1243839600000000000

    snapshots_Dgbw1,user=123abc duration=7979,productivity=0.24,pSeconds=1.91496 1243879200000000000

 second one causing the following panic

```
[http] 2015/06/11 21:17:42 79.228.245.71 - - [11/Jun/2015:21:17:42 +0200] POST /write?db=myproject-test HTTP/1.1 401 50 - - 88a58b6f-106e-11e5-8003-000000000000 67.704µs
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x66955f]

goroutine 23 [running]:
github.com/influxdb/influxdb/influxql.(*SelectStatement).RewriteDistinct(0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/ast.go:780 +0x37f
github.com/influxdb/influxdb/tsdb.(*QueryExecutor).rewriteSelectStatement(0xc20808f740, 0x0, 0xc2080fc798, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:271 +0x18a
github.com/influxdb/influxdb/tsdb.(*QueryExecutor).executeSelectStatement(0xc20808f740, 0x0, 0xc2080720c0, 0xc2081700bc, 0x10, 0xc208122840, 0x2710, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:219 +0x58
github.com/influxdb/influxdb/tsdb.func·004()
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:157 +0x727
created by github.com/influxdb/influxdb/tsdb.(*QueryExecutor).ExecuteQuery
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:211 +0x22b

goroutine 1 [chan receive (nil chan)]:
main.(*Main).Run(0xc20817af60, 0xc20800a010, 0x4, 0x4, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:81 +0x450
main.main()
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:39 +0x1d0

goroutine 6 [syscall]:
os/signal.loop()
        /root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1

goroutine 8 [IO wait]:
net.(*pollDesc).Wait(0xc2080103e0, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2080103e0, 0x0, 0x0)
        /root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208010380, 0x0, 0x7f3413357cc8, 0xc2080f2000)
        /root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc208038648, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc208038648, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc2080b9b30, 0x7f341335a068, 0xc208038648, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:45 +0x57
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:274 +0x762
```

**Comment by neonstalwart** [comment]
this looks to be the same stack trace as the panic reported in #2932.  if i'm not mistaken, the logs indicate a successful write followed by a failure for the next query - which is how it can be the same issue as #2932 (which is a query).

**Comment by otoolep** [comment]
Can you show the entire sequence @consense -- creating the database, retention policy if one was was created, and the entire `curl` command that causes the panic?

The trace you are showing is actually a query, not a write. Were you generating query load at the time?

**Comment by consense** [comment]
It looks like the problem originated from a drop measurement call before the 2 writes and the subsequent select. This is part of our CI-runs which scaffold-teardown in quick succession.
Is the http return from DROP MEASUREMENT now async? It wasnt in 0.8 as far as I recall

**Comment by fromYukki** [comment]
I have the same problem. But I notices that this problem came after I delete and recteated new database.

**Comment by consense** [comment]
maybe related to #2955 ?

**Comment by jonaz** [comment]
@consense how could it be? #2955 does not produce any panics.

**Comment by beckettsean** [comment]
Possibly fixed by #2955. @consense can you try to repro from master?

**Comment by otoolep** [comment]
Trace is from a query, not from a write, so the write data is not applicable. The crash that is shown has been fixed by the fixed in 0.9.1 (which will be released within a few days).

I'm going to close this, but if you can reproduce a crash with the write data, please open a new ticket.

Thanks @consense

**Comment by consense** [comment]
Sorry for the late response. 
I haven't been able to reproduce the above panic with Master. Thanks a bunch!
